modal/_function_utils.py

```diff
@@ -94,17 +102,20 @@ def __init__(self, f, serialized=False):
             self.definition_type = api_pb2.Function.DEFINITION_TYPE_FILE
             self.is_package = False
             self.is_file = True
+            self.serialized_function = None
         else:
             self.module_name = None
             self.base_dir = os.path.abspath("")  # get current dir
             self.definition_type = api_pb2.Function.DEFINITION_TYPE_SERIALIZED
             self.is_package = False
             self.is_file = False
+            self.serialized_function = cloudpickle.dumps(self.raw_f)
+            logger.debug(f"Serializing {self.raw_f.__qualname__}, size is {len(self.serialized_function)}")
 
         if self.definition_type == api_pb2.Function.DEFINITION_TYPE_FILE:
             # Sanity check that this function is defined in global scope
             # Unfortunately, there's no "clean" way to do this in Python
-            if "<locals>" in self.function_name.split("."):
+            if not is_global_function(f.__qualname__):
                 raise InvalidError("Modal can only import functions defined in global scope")
```

Comment:
shouldn't this be `LocalFunctionError` too?

Reply:
separately, we _could_ let people know of `serialized=True` here? maybe? i don't know